components/sup/src/http_gateway.rs

```diff
@@ -198,6 +265,25 @@ impl Server {
     }
 }
 
+fn tls_config(tls_files: Option<(path::PathBuf, path::PathBuf)>) -> Option<ServerConfig> {
+    match tls_files {
+        Some((key_file, cert_file)) => {
+            let mut config = ServerConfig::new(NoClientAuth::new());
+            let key_file = &mut BufReader::new(File::open(key_file).unwrap());
+            let cert_file = &mut BufReader::new(File::open(cert_file).unwrap());
+
+            // The validation for both the key file and the cert file happen upfront when we parse
+            // CLI args. By the time we get to this place, we know for certain that we have
+            // valid key and cert files, so we unwrap everything here.
+            let cert_chain = certs(cert_file).unwrap();
+            let mut keys = rsa_private_keys(key_file).unwrap();
+            config.set_single_cert(cert_chain, keys.remove(0)).unwrap();
```

Comment:
I don't think these `unwrap`s are safe. [Peeking into the source](https://docs.rs/rustls/0.14.0/src/rustls/pemfile.rs.html), they can fail due to read errors, which I think we'd prefer to handle better than panicking. It's fine to use `?` to propagate those errors to a caller, though. It's a bummer that `rustls` swallows the actual errors and returns only `()`, but there's not much we can do about that.

Also, I'd suggest modifying the `use` statement to just `rustls::internal::pemfile` and making these calls `pemfile::certs` and `pemfile::rsa_private_keys`. That's the more common rust idiom for external crates since doing it this way makes it easy to confuse those functions with crate-local ones.

Reply:
The main issue I had was how to propagate the errors back to the calling thread in an ergonomic way.  I'll work on this some more.